docs/architecture/dapr-for-net-developers/reference-application.md

```diff
@@ -52,38 +52,606 @@ As focus of the eShopOnDapr reference application is on Dapr, the original appli
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: pubsub
+  namespace: default
+spec:
+  type: pubsub.nats
+  version: v1
+  metadata:
+  - name: natsURL
+    value: nats://demo.nats.io:4222
+```
+
+The preceding configuration specifies the desired [NATS message broker](https://nats.io/) for this example. To change message brokers, you need only to configure a different message broker, such as RabbitMQ or Azure Service Bus and update the yaml file. With Dapr, there are no changes to your mainline service code when switching message brokers.
+
+Finally, you might ask, "Why would I need multiple message brokers in an application?". Many times a system will handle workloads with different characteristics. One event may occur 10 times a day, but another event occurs 5,000 times per second. You may benefit by partitioning messaging traffic to different message brokers. With Dapr, you can add multiple pub/sub component configurations, each with a different name.
+
+### Bindings
+
+eShopOnDapr uses the bindings building block for sending e-mails. When a user places an order, the application sends an order confirmation e-mail using the [SendGrid](https://docs.dapr.io/operations/components/setup-bindings/supported-bindings/sendgrid/) output binding. You can find this binding in the `eshop-email.yaml` file in the components folder:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: sendmail
+  namespace: eshop
+spec:
+  type: bindings.twilio.sendgrid
+  version: v1
+  metadata:
+  - name: apiKey
+    secretKeyRef:
+      name: sendGridAPIKey
+auth:
+  secretStore: eshop-secretstore
+```
+
+This configuration uses the [Twilio SendGrid](https://github.com/dapr/components-contrib/tree/master/bindings/twilio) binding component. Note how the API key for connecting to the service consumes a Dapr secret reference. This approach keeps secrets outside of the configuration file. Read the [secrets building block chapter](secrets.md) to learn more about Dapr secrets.
+
+The binding configuration specifies a binding component that can be invoked using the `/sendmail` endpoint on the Dapr sidecar. Here's a code snippet in which an email is sent whenever an order is started:
+
+```csharp
+public Task Handle(OrderStartedDomainEvent notification, CancellationToken cancellationToken)
+{
+    var string message = CreateEmailBody(notification);
+    var metadata = new Dictionary<string, string>
+    {
+        {"emailFrom", "eShopOn@dapr.io"},
+        {"emailTo", notification.UserName},
+        {"subject", $"Your eShopOnDapr order #{notification.Order.Id}"}
+    };
```

Comment:
```suggestion
    {
        ["emailFrom"] = "eShopOn@dapr.io",
        ["emailTo" = notification.UserName,
        ["subject"] = $"Your eShopOnDapr order #{notification.Order.Id}"
    };
```

For consistency with the other dictionary collection initializer usages in these articles.